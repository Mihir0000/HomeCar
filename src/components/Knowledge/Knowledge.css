.knowledge_section {
    width: 100%;
    float: left;
    background-image: url(../../../public/static/img/knowledge-bg.png);
    background-size: 100%;
    height: 40vmax;
    background-repeat: no-repeat;
    padding-top: 180px;
    padding-bottom: 240px;
}


.layout_padding {
    padding-top: 90px;
    padding-bottom: 0px;
}

.knowledge_main {
    width: 100%;
    float: left;

}

.left_main {
    width: 70%;
    float: left;
}

.right_main {
    width: 30%;
    float: left;
}

.knowledge_taital {
    width: 100%;
    float: left;
    font-size: 40px;
    color: #ffffff;
    font-weight: bold;
}

.knowledge_text {
    width: 100%;
    float: left;
    font-size: 16px;
    color: #ffffff;
    margin: 0px;
}

.play_icon {
    width: 130px;
    float: left;
    position: relative;
    z-index: 5;
}

.play_icon::before {
    content: '';
    position: absolute;
    top: 50%;
    left: 50%;
    width: 100%;
    height: 100%;
    transform: translate(-50%, -50%);
    background-color: #ffffff;
    border-radius: 100%;
    z-index: -1;
    animation: fadeEffect 1.3s infinite ease;
    box-shadow: 0px 0px 10px 0px;
}

@keyframes fadeEffect {
    0% {
        transform: translate(-50%, -50%) scale(1);
        opacity: 1
    }
    100% {
        transform: translate(-50%, -50%) scale(1.2);
        opacity: 0;
    }
}
@media screen and (max-width:678px){
    .knowledge_section{
        height: 70vmax;
    }
}