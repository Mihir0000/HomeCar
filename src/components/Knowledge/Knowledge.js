import React from 'react';
import { Card, Col, Container, Image, Row } from 'react-bootstrap';
import './Knowledge.css';

const Knowledge = () => {
    return (
        <Container fluid className="knowledge_section layout_padding">
            <Container>
                <Row className="knowledge_main">
                    <Col className="left_main">
                        <Card.Title
                            className="knowledge_taital"
                            style={{ fontSize: '40px' }}
                        >
                            Knowledge of center
                        </Card.Title>
                        <Card.Text className="knowledge_text">
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit, sed do eiusmod tempor incididunt ut labore et
                            dolore magna aliqua. Ut enim ad minim veniam
                        </Card.Text>
                    </Col>
                    <Col className="right_main d-flex align-item-center justify-content-center">
                        <Card
                            className="play_icon bg-transparent"
                            style={{ border: 'none' }}
                        >
                            <Card.Link href="/#">
                                <Image
                                    src="/static/img/play-icon.png"
                                    alt="play"
                                />
                            </Card.Link>
                        </Card>
                    </Col>
                </Row>
            </Container>
        </Container>
    );
};

export default Knowledge;
