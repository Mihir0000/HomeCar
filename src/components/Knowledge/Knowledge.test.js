import React from 'react';
import { render } from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Knowledge from './Knowledge';

test('snapshot testing --knowledge', () => {
    const component = renderer.create(<Knowledge />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

// crash test --Knowledge
it('render without crashing --knowledge', () => {
    const div = document.createElement('div');
    render(<Knowledge />, div);
    ReactDOM.unmountComponentAtNode(div);
});
