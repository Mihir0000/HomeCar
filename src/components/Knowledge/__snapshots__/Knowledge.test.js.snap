// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing --knowledge 1`] = `
<div
  className="knowledge_section layout_padding container-fluid"
>
  <div
    className="container"
  >
    <div
      className="knowledge_main row"
    >
      <div
        className="left_main col"
      >
        <div
          className="knowledge_taital card-title h5"
          style={
            Object {
              "fontSize": "40px",
            }
          }
        >
          Knowledge of center
        </div>
        <p
          className="knowledge_text card-text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam
        </p>
      </div>
      <div
        className="right_main d-flex align-item-center justify-content-center col"
      >
        <div
          className="play_icon bg-transparent card"
          style={
            Object {
              "border": "none",
            }
          }
        >
          <a
            className="card-link"
            href="/#"
          >
            <img
              alt="play"
              className=""
              src="/static/img/play-icon.png"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
